### ISSUE TITLE
If WebSocket receive buffer splits message in the middle of UTF8 character error is thrown

### ISSUE THREAD

**Issue by BrennanConroy** [mono/mono#14339]
This bug was fixed in CoreFX via https://github.com/dotnet/corefx/pull/29840

## Steps to Reproduce

TLDR: I've attached a repro project below these steps

1. Create a websocket connection to a server that will echo back (wss://echo.websocket.org is a convenient public endpoint for this)
2. Send a single multi-byte character (send as a text frame), I used "\u00E6"
3. Call `ReceiveAsync` with a buffer of size 1
4. Observe failure

**Repro project**: [Websockets.zip](https://github.com/mono/mono/files/3143134/Websockets.zip)

### Current Behavior

If you pass a buffer to `WebSocket.ReceiveAsync` and the buffer isn't big enough to hold the entire payload and the end of the buffer contains a single byte from a multi-byte character then it will throw.

### Expected Behavior

The buffer should be filled up and returned correctly, and the user will call `ReceiveAsync` again with another buffer to get the rest of the payload.

## On which platforms did you notice this

[ ] macOS
[ ] Linux
[X] Windows (only tried Windows)

**Version Used**: Xamarin 16.1.0.537

**Comment by MaximLipnin** [comment]
I reproduced the issue in VS2019 16.0.3 (windows) with the latest Xamarin.Android 9.2.3 which uses Mono 5.18 (2018-08).
Mono's CoreFX fork has the mentioned fix for this issue starting from Mono 6.0 [2019-02](https://github.com/mono/corefx/blob/mono-2019-02/src/Common/src/System/Net/WebSockets/ManagedWebSocket.cs#L733).

**Comment by steveisok** [comment]
@MaximLipnin If you didn't already, can you validate against 2019-02? Then once that's released we can close this issue.

**Comment by MaximLipnin** [comment]
@steveisok Sure, I'll validate it once again.

**Comment by MaximLipnin** [comment]
The code from the attached sample project works for me with 2019-02:
```
using System;
using System.Net.WebSockets;
using System.Text;
using System.Threading.Tasks;

class C
{
    // https://github.com/mono/mono/issues/14339
    static async Task Main()
    {
        var socket = new ClientWebSocket();
        await socket.ConnectAsync(new Uri("wss://echo.websocket.org"), default);

        var buffer = Encoding.UTF8.GetBytes("\u00E6");
        await socket.SendAsync(new ArraySegment<byte>(buffer), WebSocketMessageType.Text, true, default);

        // Changing this to 2 will make it not throw
        var buf = new byte[1];
        try
        {
            while (socket.State == WebSocketState.Open)
            {
                var res = await socket.ReceiveAsync(new ArraySegment<byte>(buf), default);
                Console.WriteLine(res.Count);
            }
        }
        catch (Exception ex)
        {
            // An Exception should not be thrown
            Console.WriteLine(ex);
        }
    }
}
```
Its output is:
```
1
1
```

**Comment by steveisok** [comment]
Since the issue is fixed in 2019-02, we can close.
